src/hangar/checkout.py

```diff
@@ -451,18 +475,12 @@ def __acquire_writer_lock(self):
         try:
             heads.acquire_writer_lock(self._branchenv, self._writer_lock)
         except PermissionError as e:
-            try:
+            with suppress(AttributeError):
```

Comment:
same as above; may use the same `with` statement? and perhaps delete other attributes as well

Reply:
see https://github.com/tensorwerk/hangar-py/pull/94#discussion_r310359814

Also I don't want to delete other attributes since the object should still be "`introspect`-able". We just need to delete any attributes which could access data on disk.